homework/App.js

```diff
@@ -14,16 +14,62 @@ class App {
   async initialize(url) {
     // Add code here to initialize your app
     // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
+    this.root = document.getElementById('root');
 
-    const root = document.getElementById('root');
+    // header:
+    const header = Util.createAndAppend('header', this.root, { class: 'header' });
 
-    Util.createAndAppend('h1', root, { text: 'It works!' }); // TODO: replace with your own code
+    // title:
+    Util.createAndAppend('h1', header, { text: 'HYF Repositories' });
+
+    // select element:
+    const select = Util.createAndAppend('select', header);
+
+    // the container:
+    this.theContainer = Util.createAndAppend('div', this.root, { id: 'container' });
```

Comment:
you don't need to set everything on the this keyword, unless you have to use them in your class again somewhere.
so you may need to have this as a normal variable, while some of it's fellows can be on the `this`. 🙂